clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -141,10 +141,8 @@ public void ensureFreshMetadata() {
      * on the current poll if one is active, or the next poll.
      */
     public void wakeup() {
-        if (wakeupsEnabled) {
-            this.wakeup.set(true);
-            this.client.wakeup();
-        }
+        this.wakeup.set(true);
+        this.client.wakeup();
```

Comment:
```poll(0)``` will effectively trigger the underlying ```Selector.selectNow```, which is not a blocking call and hence will not be affected by the ```Selector.wakeup``` call; instead that call will affect the ```poll(Long.MAX_VALUE)``` causing it to return immediately:

http://docs.oracle.com/javase/7/docs/api/java/nio/channels/Selector.html#wakeup()